server/store.js

```diff
@@ -33,6 +33,10 @@ function save() {
 
 var store = _.extend(new events.EventEmitter(), {
   add: function (link, callback) {
+    if (typeof link === "string")
+      if (link.match(/^(\w+)$/))
```

Comment:
I think the hash is supposed to by a certain size - 20 bytes I believe, which is 40 characters in hex.
So I would maybe convert both conditions to this: `if (/^\w{40}$/.test(link)/) ...`

Reply:
You're right. I've found however that hashes can sometimes be Base32 encoded, so I've added that case too.

Read more [here](https://stackoverflow.com/a/10961929)